frontend/app/routes/api+/switch-language.ts

```diff
@@ -1,89 +1,63 @@
 import { json } from '@remix-run/node';
-import type { ActionFunctionArgs } from '@remix-run/node';
+import type { ActionFunctionArgs, LoaderFunctionArgs } from '@remix-run/node';
 
 import { z } from 'zod';
 
+import { getInstrumentationService } from '~/services/instrumentation-service.server';
 import { getRaoidcService } from '~/services/raoidc-service.server';
 import { createLangCookie } from '~/utils/locale-utils.server';
 import { getLogger } from '~/utils/logging.server';
 
 const logger = getLogger('routes/api.switch-language');
-
-const switchLanguageDataSchema = z.object({
-  language: z.enum(['en', 'fr']),
-});
+const switchLanguageDataSchema = z.object({ language: z.enum(['en', 'fr']) });
 
 export type SwitchLanguageData = z.infer<typeof switchLanguageDataSchema>;
 
 /**
  * A loader function that always throws a 405 Method Not Allowed response.
  */
-export async function loader() {
-  logger.warn(`Method Not Allowed: The requested method (GET) is not allowed for this resource. Allowed methods: PUT.`);
-  throw json(
-    {
-      error: 'Method Not Allowed',
-      message: 'The requested method is not allowed for this resource. Allowed methods PUT.',
-    },
-    { status: 405 },
-  );
+export async function loader({ request }: LoaderFunctionArgs) {
+  logger.warn(`Method Not Allowed: The requested method (${request.method}) is not allowed for this resource. Allowed methods: PUT.`);
```

Comment:
`loader` can only be called by GET method

Reply:
I know, but I changed the log line to be consistent with the other one in the file for easier maintainability.

Reply:
Actually.. you forced me to check. (I added the IN LOADER bit to make sure it was the loader that was logging)

```
IN LOADER --- Method Not Allowed: The requested method (OPTIONS) is not allowed for this resource. Allowed methods: PUT.
```

```
IN LOADER --- Method Not Allowed: The requested method (HEAD) is not allowed for this resource. Allowed methods: PUT.
```

Reply:
Good catch!